-static YAAMP_JOB *job_get_last()
+static YAAMP_JOB *job_get_last(int coinid)
 {
        g_list_job.Enter();
        for(CLI li = g_list_job.first; li; li = li->prev)
@@ -49,6 +50,7 @@ static YAAMP_JOB *job_get_last()
                YAAMP_JOB *job = (YAAMP_JOB *)li->data;
                if(!job_can_mine(job)) continue;
                if(!job->coind) continue;
+               if(coinid > 0 && job->coind->id != coinid) continue;

                g_list_job.Leave();
                return job;
@@ -62,7 +64,8 @@ static YAAMP_JOB *job_get_last()

 void job_send_last(YAAMP_CLIENT *client)
 {
-       YAAMP_JOB *job = job_get_last();
+       YAAMP_JOB *job = job_get_last(client->coinid);
+       if(!job) job = job_get_last(0);
        if(!job) return;

        YAAMP_JOB_TEMPLATE *templ = job->templ;